tests/integration_tests/scheduler/test_lsf_driver.py

```diff
@@ -271,6 +272,7 @@ async def test_kill_before_submit_is_finished(
         job_kill_window = 5
         test_grace_time = 10
     else:
+        # More aggressive on local iron, fixing residual flakyness with reruns
```

Comment:
As discussed let's try to replace `await asyncio.sleep(0.01)` with the actual lock being set; ie. `self._submit_locks[0]`

Reply:
Turns out it that is not enough to avoid flakyness